### DISCUSSION TITLE
Multi-level router can't use empty path (eg. get.("") )

### DISCUSSION THREAD

**Discussion by jordantshaw** [fastapi/fastapi#8618] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# main.py

from fastapi import FastAPI, APIRouter
from .child import child_router

app = FastAPI()

router = APIRouter()
router.include_router(child_router, prefix='/foo')

app.include_router(router, prefix='/api/v1', include_in_schema=True)

# .child/__init__.py
from fastapi import APIRouter

from .grandchild import grandchild_router

child_router = APIRouter()
child_router.include_router(grandchild_router)

# ..grandchild/__init__.py
from fastapi import APIRouter

grandchild_router = APIRouter()

@grandchild_router.get('')
def test():
    return 'bar'
```

### Description

I currently have something similar setup as the example code.  main_router > child_router > grandchild_router.  I am setting a prefix on the child router level (during the include_router call).  When I set the prefix on the child level, I cannot use an empty path on the grandchild level router.

If I do, I got the following error:
`Exception: Prefix and path cannot be both empty (path operation: test)`

My file structure looks like this.
```
project
│   main.py (main_router)
└───child
│   │   __init__.py (child_router)
│   └───grandchild
│       │    __init__.py (grandchild_router)


### Operating System

macOS

### Operating System Details

_No response_

### FastAPI Version

0.75.0

### Python Version

Python 3.9.10

### Additional Context

_No response_

**Comment by gmanny** [comment]
Looks like routers are meant to be in a tree structure, child routers having a path inside.

This is related to the check [on this line](https://github.com/tiangolo/fastapi/blob/6125dc72cdf245573b3eb2cba66ed8d1e52ba2cd/fastapi/routing.py#L693-L696). As there is no lookup tree, and routers are really just [combined into a big routing list](https://github.com/tiangolo/fastapi/blob/6125dc72cdf245573b3eb2cba66ed8d1e52ba2cd/fastapi/routing.py#L729-L730) I would say checking [in the original route + prefix if](https://github.com/tiangolo/fastapi/blob/6125dc72cdf245573b3eb2cba66ed8d1e52ba2cd/fastapi/routing.py#L693-L696) that the router actually has an empty route first, should be easy and would relieve this situation. (Another option is checking if a route with the path exists in the [`APIRouter.add_api_route`](https://github.com/tiangolo/fastapi/blob/6125dc72cdf245573b3eb2cba66ed8d1e52ba2cd/fastapi/routing.py#L504), but this is open to all kinds of trouble - i.e. what is there are two identical paths where only variable names are different.)

**Comment by mustyoshi** [comment]
I've also just run into this issue and had to settle for having a trailing slash on the child route.

**Comment by gmanny** [comment]
I've ended up with this function, which I apply to the route router after adding all the child routers:

```python
from typing import TypeVar

from fastapi import FastAPI
from fastapi.routing import APIRoute, APIRouter

ParentT = TypeVar("ParentT", APIRouter, FastAPI)


def remove_trailing_slashes_from_routes(parent: ParentT) -> ParentT:
    "Removes trailing slashes from all routes in the given router"

    for route in parent.routes:
        if isinstance(route, APIRoute):
            route.path = route.path.rstrip("/")

    return parent
```

**Comment by JoshYuJump** [comment]
Same issue here.
